crates/typst/src/geom/color.rs

```diff
@@ -1,17 +1,105 @@
 use ecow::{eco_format, EcoString};
+use palette::encoding::{self, Linear};
+use palette::{Darken, Desaturate, FromColor, Lighten, RgbHue, Saturate, ShiftHue};
+use std::f64::consts::TAU;
 use std::str::FromStr;
+use typst_syntax::Spanned;
 
 use super::*;
 use crate::diag::{bail, At, SourceResult};
 use crate::eval::{cast, Args, Array, Cast, Func, Str};
-use crate::syntax::Spanned;
+
+// Type aliases for `palette` internal types in f64.
+type Oklab = palette::oklab::Oklaba<f64>;
+type LinearRgba = palette::rgb::Rgba<Linear<encoding::Srgb>, f64>;
+type Rgba = palette::rgb::Rgba<encoding::Srgb, f64>;
+type Hsl = palette::hsl::Hsla<encoding::Srgb, f64>;
+type Hsv = palette::hsv::Hsva<encoding::Srgb, f64>;
+type Luma = palette::luma::Luma<encoding::Srgb, f64>;
+
+/// A trait containing all common functionality of colors.
+pub trait ColorExt: Sized + Copy + Into<Color> {
```

Comment:
I think the `Color` type itself should come first as the top thing in the module.